Gcd_nobis-julian/src/gcd_nobis/julian/Gcd_nobisJulian.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * This might not be the most elegant solution out there
+ */
+package gcd_nobis.julian;
+
+/**
+ *
+ * @author Julian Nobis
+ */
+public class Gcd_nobisJulian {
+    static int number1 = 1989, number2 = 867; // exact same numbers like in school
+    static int number3 = -3, number4 = 111; // should get a message because number3 is negative
+    public static void main(String[] args) {
+        System.out.println("This is the solution of Julian Nobis\n");
+        eucledianStructure();
+        primeFactorsStructure();
+    } 
+    // method 1: eucledian algorithm
+    public static void eucledianStructure(){
+        int resultEucledian = gcdEucledian(number1, number2);
+        output(1, resultEucledian);
+    }
+    // method 2: prime-factors algorithm
+    public static void primeFactorsStructure(){
+        int resultPrimeFactors = gcdPrimeFactors(number1, number2);
+        output(2, resultPrimeFactors);
+    }
+    
+    public static void output(int number, int result){

```

Comment:
Here I would suggest the first parameter named gcdCalculationStyle being of type enum (GcdCalculationStyle {CLASSICAL, EUCLIDEAN}).